orca-queue-sql/src/test/kotlin/com/netflix/spinnaker/orca/q/sql/SqlQueueIntegrationTest.kt

```diff
@@ -0,0 +1,178 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.q.sql
+
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.module.SimpleModule
+import com.fasterxml.jackson.module.kotlin.KotlinModule
+import com.netflix.spectator.api.Registry
+import com.netflix.spinnaker.config.ObjectMapperSubtypeProperties
+import com.netflix.spinnaker.config.SpringObjectMapperConfigurer
+import com.netflix.spinnaker.config.SqlProperties
+import com.netflix.spinnaker.config.TransactionRetryProperties
+import com.netflix.spinnaker.kork.dynamicconfig.DynamicConfigService
+import com.netflix.spinnaker.kork.jedis.RedisClientDelegate
+import com.netflix.spinnaker.kork.jedis.RedisClientSelector
+import com.netflix.spinnaker.kork.sql.config.SqlRetryProperties
+import com.netflix.spinnaker.kork.sql.test.SqlTestUtil
+import com.netflix.spinnaker.orca.TaskResolver
+import com.netflix.spinnaker.orca.config.JedisConfiguration
+import com.netflix.spinnaker.orca.config.RedisConfiguration
+import com.netflix.spinnaker.orca.pipeline.model.Execution
+import com.netflix.spinnaker.orca.pipeline.persistence.ExecutionRepository
+import com.netflix.spinnaker.orca.q.QueueIntegrationTest
+import com.netflix.spinnaker.orca.q.TestConfig
+import com.netflix.spinnaker.orca.q.migration.ExecutionTypeDeserializer
+import com.netflix.spinnaker.orca.q.migration.TaskTypeDeserializer
+import com.netflix.spinnaker.orca.q.sql.pending.SqlPendingExecutionService
+import com.netflix.spinnaker.orca.sql.pipeline.persistence.SqlExecutionRepository
+import com.netflix.spinnaker.orca.test.redis.EmbeddedRedisConfiguration
+import com.netflix.spinnaker.q.Queue
+import com.netflix.spinnaker.q.metrics.EventPublisher
+import com.netflix.spinnaker.q.metrics.MonitorableQueue
+import com.netflix.spinnaker.q.sql.SqlQueue
+import de.huxhorn.sulky.ulid.ULID
+import org.jooq.DSLContext
+import org.junit.runner.RunWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.test.context.junit4.SpringRunner
+import java.time.Clock
+import java.time.Duration
+import java.util.Optional
+
+@Configuration
+class SqlTestConfig {
+  @Bean
+  fun jooq(): DSLContext {
+    val testDatabase = SqlTestUtil.initTcMysqlDatabase()
+    return testDatabase.context
+  }
+
+  @Autowired
+  fun sqlQueueObjectMapper(
```

Comment:
I'm confused, is this configuring an existing bean or creating one?

Reply:
Configuring existing, the same thing is happening here for the redis equivalent via: https://github.com/spinnaker/orca/blob/master/orca-queue-redis/src/main/kotlin/com/netflix/spinnaker/config/RedisOrcaQueueConfiguration.kt#L51-L56

Reply:
Hmm, but this is changing the mapper for everything then, right?

Reply:
Yeah, seems to be required for orca to function. Are you suggesting this should happen other than Queue specific configuration classes?

Reply:
I'm just wondering if that configuration is Keiko-specific it might cause unexpected behavior elsewhere. I'm also surprised that `KotlinModule`, etc. aren't already configured.

I don't think it's a _big_ deal though.

Reply:
It might be nice wiring in `ExecutionTypeDeserializer` and `TaskTypeDeserializer` in `OrcaObjectMapper` to make it more obvious that they're globally applied but they aren't currently available from orca-core and I'm hesitant to change that.